solr/core/src/java/org/apache/solr/handler/admin/InfoHandler.java

```diff
@@ -157,4 +158,17 @@ public SolrRequestHandler getSubHandler(String subPath) {
   public Boolean registerV2() {
     return Boolean.TRUE;
   }
+
+  @Override
+  public Name getPermissionName(AuthorizationContext request) {
+    // Delegate permission to the actual handler
+    String path = request.getResource();
+    String lastPath = path.substring(path.lastIndexOf("/") +1 );
+    RequestHandlerBase handler = handlers.get(lastPath.toLowerCase(Locale.ROOT));
+    if (handler != null) {
+      return handler.getPermissionName(request);
+    } else {
+      return null;
```

Comment:
[Q] What does 'null' mean as a return value from `PermissionNameProvider.getPermissionName`?

Reply:
Good question. I used it as a fallback in line with what @noblepaul has used e.g. here https://github.com/apache/solr/blob/main/solr/core/src/java/org/apache/solr/handler/SchemaHandler.java#L112 but I have not tested what it actually does. In this case I think this path will never be reached.